internal/controllers/machine/machine_controller_status_test.go

```diff
@@ -1128,12 +1128,12 @@ func TestDeletingCondition(t *testing.T) {
 			},
 			reconcileDeleteExecuted: true,
 			deletingReason:          clusterv1.MachineDeletingWaitingForPreDrainHookV1Beta2Reason,
-			deletingMessage:         "Waiting for pre-drain hooks to complete (hooks: test-hook)",
+			deletingMessage:         "Waiting for pre-drain hooks to succeed (hooks: test-hook)",
```

Comment:
Q:what about adding a test with more than one hook validating that hooks are sorted?

Reply:
These tests are not executing the fixed code

Reply:
We have no code coverage on the pre-drain code path (this PR) or the pre-terminate code path (fixed in the previous PR: #11487).

I can open a follow-up issue if you want, but I wouldn't want to block this PR because of that if possible